plugins/text-editor-resources/src/components/extension/inlineComment.ts

```diff
@@ -0,0 +1,984 @@
+//
+// Copyright © 2024 Hardcore Engineering Inc.
+//
+// Licensed under the Eclipse Public License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License. You may
+// obtain a copy of the License at https://www.eclipse.org/legal/epl-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import { getResource } from '@hcengineering/platform'
+import { type Editor, Mark } from '@tiptap/core'
+import {
+  type EditorState,
+  Plugin,
+  PluginKey,
+  type PluginView,
+  type Selection,
+  TextSelection,
+  type Transaction
+} from '@tiptap/pm/state'
+import { Decoration, DecorationSet, type EditorView } from '@tiptap/pm/view'
+import { SvelteRenderer } from '../node-view'
+import type { AnyComponent, AnySvelteComponent } from '@hcengineering/ui'
+import { Fragment, Slice, type Node } from '@tiptap/pm/model'
+import { type Account, type Markup, type Ref, type Timestamp, getCurrentAccount, generateId } from '@hcengineering/core'
+import tippy, { type Instance } from 'tippy.js'
+import 'tippy.js/animations/shift-toward.css'
+import { type Doc as YDoc, type Map as YMap } from 'yjs'
+import core from '@hcengineering/core'
+
+interface InlineCommentExtensionOptions {
+  boundary?: HTMLElement
+  popupContainer?: HTMLElement
+  commentWidth?: number
+
+  ydoc: YDoc
+  ydocExtensionField?: string
+
+  minEditorWidth?: number
+
+  requestSideSpace?: (width: number) => void
+}
+
+type InlineCommentDisplayMode = 'compact' | 'full'
+
+interface PointerState {
+  hover: Set<string>
+  focus: Set<string>
+}
+
+interface MetaPatch {
+  newCommentRequested?: boolean
+  displayModeConstraint?: InlineCommentDisplayMode
+  component?: AnySvelteComponent
+  pointer?: Partial<PointerState>
+  threads?: Map<string, Thread>
+}
+
+function getMeta (tx?: Transaction): MetaPatch | undefined {
+  return tx?.getMeta('inline-comment')
+}
+
+function setMeta (tx: Transaction, meta: MetaPatch): Transaction {
+  return tx.setMeta('inline-comment', meta).setMeta('addToHistory', false)
+}
+
+function getYDocCommentMap (options: InlineCommentExtensionOptions): YMap<InlineComment> {
+  return options.ydoc.getMap(options.ydocExtensionField ?? 'inline-comments')
```

Comment:
this won't work :( we currently store only one field from ydoc. we will need to do a bit more changes to allow to store additional information

Reply:
as discussed on the call, keep this as is for now